### PR TITLE
feat: Add a per group toggle for autobalance

### PR DESCRIPTION
**Pull request by hauspie**
This pull request adds commands for a player to enable or disable the autobalance system on a per group basis.

Dungeon and raids autobalance will now be enabled or disabled based on the configuration of the party leader. This allows player to choose either they want autobalance or not (kind of similar to the normal/heroic mode choice)

This functionnality can be enabled or disabled in the config file so that if you want your server to force autobalance for everyone, you still can.

### CODE REVIEW

File: src/AutoBalance.cpp [hunk 1]
```diff
@@ -110,10 +118,43 @@ static std::map<int, int> forcedCreatureIds;
...
 InflectionPoint, InflectionPointRaid, InflectionPointRaid10M, InflectionPointRaid25M, InflectionPointHeroic, InflectionPointRaidHeroic, InflectionPointRaid10MHeroic, InflectionPointRaid25MHeroic, BossInflectionMult;
 
+
+static Player *GetGroupLeaderOrSelf(Player *player)
+{
+    Group *group = player->GetGroup();
+    Player *groupLeader = player;
+
+    // If player is in a group, get the group leader
+    if (NULL != group)
```
Comment (Winfidonarleyan) [inline src/AutoBalance.cpp:132]:
```cpp
if (group)
```

Comment (Helias) [reply src/AutoBalance.cpp:132]: same for `if (NULL != leader)` I guess

File: src/AutoBalance.cpp [hunk 2]
```diff
@@ -801,14 +853,72 @@ class AutoBalance_CommandScript : public CommandScript
...
+        AutoBalancePlayerInfo *playerABInfo = groupLeader->CustomData.GetDefault<AutoBalancePlayerInfo>("AutoBalancePlayerInfo");
+        // TODO: should use database to store localized strings
+        if (player->GetSession()->GetSessionDbcLocale() == LOCALE_frFR)
+            handler->PSendSysMessage("L'équilibrage automatique est %s.", playerABInfo->enabled ? "activé" : "désactivé");
+        else
+            handler->PSendSysMessage("Autobalance is %s.", playerABInfo->enabled ? "enabled" : "disabled");
+        return true;
+    }
+
+    static bool SetABAutobalance(ChatHandler *handler, bool enable)
```
Comment (Yehonal) [inline src/AutoBalance.cpp:888]: Maybe this should be named something like:

```suggestion
    static bool SetGroupAutobalance(ChatHandler *handler, bool enable)
```

**Review by Yehonal — CHANGES_REQUESTED** [review]
This is an interesting feature @hauspie can you complete this with our suggestions?

**Comment by pangolp** [comment]
Greetings. What is the current status of this pull request? I ask, because the idea is to try to keep the repository as updated as possible, and I see that it was created in 2020, and has not had recent messages. In addition, it has some conflicts. In any case, you can close and take the code, with the respective credits and create a new application. I am waiting for a confirmation message. Thank you.

**Pull request closed without merging**